clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -272,6 +272,7 @@ private boolean trySend(long now) {
 
     private void clientPoll(long timeout, long now) {
         client.poll(timeout, now);
+        now = time.milliseconds();
```

Comment:
If we introduce a new interface, we should be careful to use the right time source (see KAFKA-2607).